src/test/java/com/pgs/client/service/AuthenticationClientTest.java

```diff
@@ -42,32 +42,30 @@
 
     @BeforeEach
     void testGetTokenSetUp() {
-        setAuthenticationClientFields();
-        when(authRestTemplate.postForObject(TOKEN_URL, setUpRequest(), Token.class))
-                .thenReturn(TOKEN_DTO);
+        ReflectionTestUtils.setField(authenticationClient, "username", USERNAME);
+        ReflectionTestUtils.setField(authenticationClient, "password", PASSWORD);
+        ReflectionTestUtils.setField(authenticationClient, "grantType", GRANT_TYPE);
+        ReflectionTestUtils.setField(authenticationClient, "tokenUrl", TOKEN_URL);
+        when(authRestTemplate.postForObject(TOKEN_URL, setUpRequest(GRANT_TYPE), Token.class))
+                .thenReturn(TOKEN);
     }
 
     @Test
     void testGetToken() {
+        when(authRestTemplate.postForObject(TOKEN_URL, setUpRequest(GRANT_TYPE), Token.class))
+                .thenReturn(TOKEN);
         var token = authenticationClient.getToken();
-        verify(authRestTemplate).postForObject(TOKEN_URL, setUpRequest(), Token.class);
-        assertEquals(TOKEN_DTO.getAccessToken(), token.getAccessToken());
-        assertEquals(TOKEN_DTO.getTokenType(), token.getTokenType());
-        assertEquals(TOKEN_DTO.getRefreshToken(), token.getRefreshToken());
-        assertEquals(TOKEN_DTO.getExpiresIn(), token.getExpiresIn());
-        assertEquals(TOKEN_DTO.getScope(), token.getScope());
-    }
-
-    private void setAuthenticationClientFields() {
-        ReflectionTestUtils.setField(authenticationClient, "username", USERNAME);
-        ReflectionTestUtils.setField(authenticationClient, "password", PASSWORD);
-        ReflectionTestUtils.setField(authenticationClient, "grantType", GRANT_TYPE);
-        ReflectionTestUtils.setField(authenticationClient, "tokenUrl", TOKEN_URL);
+        verify(authRestTemplate).postForObject(TOKEN_URL, setUpRequest(GRANT_TYPE), Token.class);
+        assertEquals(TOKEN.getAccessToken(), token.getAccessToken());
+        assertEquals(TOKEN.getTokenType(), token.getTokenType());
+        assertEquals(TOKEN.getRefreshToken(), token.getRefreshToken());
+        assertEquals(TOKEN.getExpiresIn(), token.getExpiresIn());
+        assertEquals(TOKEN.getScope(), token.getScope());
```

Comment:
Why don't you just check if one token is equal to another? If you check all the fields checking everything on such atomic level doesn't make sense.